packages/advanced-logic/specs/payment-network-erc20-fee-proxy-contract-0.1.0.md

```diff
@@ -0,0 +1,328 @@
+
+|  Property                     |  Value                              |
+| ----------------------------- | ----------------------------------- |
+| **name**                      | Constant value: "addPaymentAddress" |
+| **parameters**                |                                     |
+| **parameters.paymentAddress** | `paymentAddress` from parameters    |
+
+#### addRefundAddress
+
+##### Parameters
+
+|                              | Type   | Description                                   | Requirement   |
+| ---------------------------- | ------ | --------------------------------------------- | ------------- |
+| **id**                       | String | Constant value: "pn-erc20-fee-proxy-contract" | **Mandatory** |
+| **action**                   | String | Constant value: "addRefundAddress"            | **Mandatory** |
+| **parameters**               | Object |                                               |               |
+| **parameters.refundAddress** | String | Ethereum address for the refund               | **Mandatory** |
+
+##### Conditions
+
+This action is valid if:
+
+- The extension state with the id "pn-erc20-fee-proxy-contract" exists
+- The signer is the `payer`
+- The extension property `refundAddress` is undefined
+
+##### Warnings
+
+None.
+
+##### Results
+
+An extension state is updated with the following properties:
+
+|  Property                |  Value                                                 |
+| ------------------------ | ------------------------------------------------------ |
+| **values.refundAddress** | `refundAddress` from parameters                        |
+| **events**               | Add an 'addRefundAddress' event (see below) at its end |
+
+The 'addRefundAddress' event:
+
+|  Property                    |  Value                          |
+| ---------------------------- | ------------------------------- |
+| **name**                     | 'addRefundAddress'              |
+| **parameters**               |                                 |
+| **parameters.refundAddress** | `refundAddress` from parameters |
+
+#### addFeeAddress
+
+##### Parameters
+
+|                           | Type   | Description                                   | Requirement   |
+| ------------------------- | ------ | --------------------------------------------- | ------------- |
+| **id**                    | String | Constant value: "pn-erc20-fee-proxy-contract" | **Mandatory** |
+| **action**                | String | Constant value: "addFeeAddress"               | **Mandatory** |
+| **parameters**            | Object |                                               |               |
+| **parameters.feeAddress** | String | Ethereum address for the fee payment          | **Mandatory** |
+
+##### Conditions
+
+This action is valid, if:
+
+- The extension state with the id "pn-erc20-fee-proxy-contract" exists
+- The signer is the `payee`
+- The extension property `feeAddress` is undefined
+
+##### Warnings
+
+None.
```

Comment:
There should be a warning if the payer tries to perform this action

Reply:
Same as previous.